gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/client/ClientUrlUtils.java

```diff
@@ -23,6 +23,9 @@
  * Helper implementation which wraps calls to code which require a running GWT environment and make testing slow.
  */
 public class ClientUrlUtils implements UrlUtils {
+    // TODO: Review what should be encoded in various parameter types. Don't rely on GWT's code.
```

Comment:
The behavior is correct for query string even if it encodes more than actually required. The extra characters encoded are actually those needed when encoding matrix params. pathSegment however encodes the `/`. It can be a valid character when the path template specifies a regex (ie `@Path("/books/{book:.+}")`). I'll have to go over it at that time.